packages/schematics/angular/web-worker/index.ts

```diff
@@ -0,0 +1,205 @@
+      }
+      const filesAstNode = findPropertyInAstObject(tsCfgAst, 'exclude');
+      if (filesAstNode && filesAstNode.kind != 'array') {
+        throw new SchematicsException('Invalid tsconfig "exclude" property; expected an array.');
+      }
+
+      if (filesAstNode && filesAstNode.value.indexOf(workerGlob) == -1) {
+        const recorder = host.beginUpdate(tsConfigPath);
+        appendValueInAstArray(recorder, filesAstNode, workerGlob);
+        host.commitUpdate(recorder);
+      }
+    }
+
+    return chain([
+      // Add tsconfigs.
+      ...tsConfigRules,
+      // Add workspace configuration.
+      updateWorkspace(workspace),
+    ]);
+  };
+}
+
+function addSnippet(options: WebWorkerOptions): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    context.logger.debug('Updating appmodule');
+
+    if (options.path === undefined) {
+      return;
+    }
+
+    const siblingModules = host.getDir(options.path).subfiles
+      // Find all files that start with the same name, are ts files, and aren't spec files.
+      .filter(f => f.startsWith(options.name) && f.endsWith('.ts') && !f.endsWith('spec.ts'))
+      // Sort alphabetically for consistency.
+      .sort();
+
+    if (siblingModules.length === 0) {
+      // No module to add in.
+      return;
+    }
+
+    const siblingModulePath = `${options.path}/${siblingModules[0]}`;
+    const workerCreationSnippet = tags.stripIndent`
+      if (typeof Worker !== 'undefined') {
+        const worker = new Worker('./${options.name}.worker', { type: 'module' });
+        worker.onmessage = ({ data }) => {
+          console.log('page got message: $\{data\}');
+        };
+        worker.postMessage('hello');
+      }
+    `;
+
+    // Append the worker creation snippet.
+    const originalContent = host.read(siblingModulePath);
+    host.overwrite(siblingModulePath, originalContent + '\n' + workerCreationSnippet);
+
+    return host;
+  };
+}
+
+export default function (options: WebWorkerOptions): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const project = getProject(host, options.project);
+    if (!options.project) {
+      throw new SchematicsException('Option "project" is required.');
+    }
+    if (!project) {
+      throw new SchematicsException(`Invalid project name (${options.project})`);
+    }
+    if (project.projectType !== 'application') {
+      throw new SchematicsException(`Web Worker requires a project type of "application".`);
+    }
+
+    if (options.path === undefined) {
+      options.path = buildDefaultPath(project);
+    }
+    const parsedPath = parseName(options.path, options.name);
+    options.name = parsedPath.name;
+    options.path = parsedPath.path;
+    const root = project.root || project.sourceRoot || '';
+
+    const templateSource = apply(url('./files/worker'), [
+      applyTemplates({ ...options, ...strings }),
+      move(parsedPath.path),
+    ]);
+
+    context.addTask(new NodePackageInstallTask());
```

Comment:
I don't see any reason why we need this, we are not adding new dependencies.

Reply:
Indeed it is not needed. Left it in as a bad copy paste. Thanks for finding this.